packages/appium/lib/appium.js

```diff
@@ -241,6 +252,330 @@ class AppiumDriver extends DriverCore {
     }
   }
 
+  /**
```

Comment:
Perhaps it would make sense to extract bidi-related stuff into a mixin class? This class is already quite big and hard to read

Reply:
sure, maybe i'll have you complete review before doing that

Reply:
just added as a TODO for a follow up PR to make it cleaner